src/Build/Evaluation/Expander.cs

```diff
@@ -4302,6 +4302,30 @@ private static bool TryGetArg(object[] args, out int arg0)
                 return TryConvertToInt(args[0], out arg0);
             }
 
+            private static bool TryGetArg(object[] args, out Version arg0)
+            {
+                if (args.Length != 1)
+                {
+                    arg0 = new Version();
+                    return false;
+                }
+
+                return TryConvertToVersion(args[0], out arg0);
+            }
+
+            private static bool TryConvertToVersion(object value, out Version arg0)
+            {
+                string val = value as string;
+
+                if (string.IsNullOrEmpty(val) || !Version.TryParse(val, out arg0))
+                {
+                    arg0 = new Version();
```

Comment:
It's worth noting this will result in arg0 being null rather than an empty value, but it shouldn't matter because the standard has been to not use arg0 if the check failed